src/model/proto/checklist.proto

```diff
@@ -13,6 +13,8 @@ message ChecklistFileMetadata {
   string aircraft_info = 5;
   string manufacturer_info = 6;
   string copyright_info = 7;
+
+  optional string description = 8;
```

Comment:
I gave a partial explanation for this in my previous comment. ForeFlight has only 3 metadata fields, all optional, that do not match the fields you have for a perfect 1:1 translation:

* Filename can sort of match "name", with the caveats in the other comment.
* Tail number can be perfectly matched 1:1
* Description doesn't obviously match any of your fields

My priority was to make sure that when I load an FF checklist and then re-export it, the metadata is as unaffected as possible. This kind of requires setting the "Name" field in the original ForeFlight checklist, but I think it's an acceptable compromise.

As for the other fields, I mapped "Tail number" 1:1 and introduced "Description" to allow 1:1 mapping. All other fields are simply ignored, which is of course suboptimal, because the information entered in them will be completely lost, unless the checklist is additionally exported in the native format.

Now there are several ways to go about this in line with your suggestions:

1. Accept my solution of adding a generic description field (and maybe implement it for other formats?)
2. Map "description" 1:1 to an existing field like "manufacturer information" instead of adding a new field (and possibly implement aggregation of all other fields into this field - the downside is that aggregated data can't be broken down properly again).

I don't have a strong opinion on this, but I have no idea about other formats, and unfortunately I'm not sure how to get a feel for how they actually display. I don't have easy access to SkyView, and they don't seem to have a simulator, and I'm not sure if I can load checklists into the Garmin simulator. So I don't know if and how other formats can benefit from a generic "description" field.

Should I implement the second option (map to "manufacturer information"), and if so, with or without aggregation? I guess that this option with aggregation is the least bad in our context?

Reply:
Here's how I'd approach this - I'd add the description field, then when loading:
- If a line starts with a known prefix like "Manufacturer info: ", map it to that field
- All lines that are left go into the description field
- For other formats that are not field-restricted (i.e. not Garmin) I can also plug in the description field in the same way

When writing:
- Output the description field first
- For each field that's filled out, output it with a prefix

That lets us connect the full functionality to most formats.

I can send a commit doing this for other formats if you think that's reasonable, then you can just follow the same.

Reply:
I think I'm mostly done now, and your suggestion sounds reasonable. I'd appreciate it if you'd add a commit for other formats, so I can implement it for ForeFlight as well.

Reply:
Most of it is already there in text-based formats: https://github.com/rdamazio/efis-editor/blob/main/src/model/formats/text-reader.ts#L259
I can add the description part to that once we merge your changes.

Reply:
Alright, I implemented the two-way translation as discussed and started testing with the app, and that's when disaster struck.

What I didn't consider is that the `description` field does not allow multiline input! And even on a single line, there are only a few dozen characters you can fit before ForeFlight truncates it. Editing the long lines is also super hard.

On the other hand, by studying the manual, I discovered that the intended use is apparently what you call "make and model". So I just mapped it that way.

<img width="740" alt="Screenshot 2024-08-26 at 09 17 15" src="https://github.com/user-attachments/assets/1fd8e574-4c12-4f8e-aac3-64303d487b28">

This leaves us with the copyright and manufacturer information being lost on ForeFlight exports, but at least everything else works as expected. What do you think, can you live with this?

Reply:
Yes, this seems fine and idiomatic. We can revise if anyone complains.